src/pyhf/optimize/autodiff.py

```diff
@@ -1,13 +1,17 @@
 """Helper Classes for use of automatic differentiation."""
-import scipy
+from scipy.optimize import minimize
```

Comment:
I don't like this. I would prefer to see `scipy.optimize.minimize` because I get confused with the existing `minimize()` defined in this file as well.